crates/core/c8y_smartrest/src/smartrest_deserializer.rs

```diff
@@ -228,6 +228,27 @@ impl SmartRestLogRequest {
     }
 }
 
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct SmartRestRestartRequest {
+    pub message_id: String,
+    pub device: String,
+}
+
+impl SmartRestRestartRequest {
+    pub fn from_smartrest(smartrest: &str) -> Result<Self, SmartRestDeserializerError> {
+        let mut rdr = ReaderBuilder::new()
+            .has_headers(false)
+            .flexible(true)
+            .from_reader(smartrest.as_bytes());
+
+        match rdr.deserialize().next() {
+            Some(Ok(record)) => Ok(record),
+            Some(Err(err)) => Err(err)?,
+            None => panic!("empty request"),
```

Comment:
This panic needs to be replace by an `Err(...)` even this cannot currently happen.

Indeed, currently this panic is safe because the `from_smartrest` method is called *only* when we have an operation starting with a `510` prefix. But this is not obvious from a local view of the code and might change.